### PR TITLE
Fallback to a text response if JSON fails

### PR DESCRIPTION
**Pull request by katspaugh**
As it turned out in https://github.com/gnosis/safe-react/pull/2615, when the gateway responds to the `/propose` request, the response is empty and thus cannot be parsed as JSON.

This PR adds a fallback to the response text.

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -29,24 +29,31 @@ export function stringifyQuery(query?: Params): string {
...
+  const resp = await fetch(url, options)
+
+  if (!resp.ok) {
+    throw Error(resp.statusText)
+  }
+
+  try {
+    return await resp.json()
+  } catch (err) {
+    return (await resp.text()) as unknown as T
```
Comment (mikheevm) [inline src/utils.ts:57]: maybe worth logging an error. Also, this should always be a string, shouldn't it?

Comment (katspaugh) [reply src/utils.ts:57]: So far the only known case where it fails is this POST request, and it's not really an error in that case. 
Let me replace the try-catch with an if here.

Comment (mikheevm) [reply src/utils.ts:57]: Maybe it makes sense to add this logic to the specific request? because this function name is `fetchJson`

Comment (katspaugh) [reply src/utils.ts]: Actually, true, the client will also expect JSON and fail. I should return an empty object instead.

Comment (mikheevm) [reply src/utils.ts]: does it make sense to change this on the backend side for consistency purposes?

Comment (katspaugh) [reply src/utils.ts]: Good idea, requested here: https://github.com/gnosis/safe-client-gateway/issues/554

**Review by mikheevm — APPROVED** [review]

**Pull request merged**